File: Sorting/Bubble Sort/BubbleSort.js
function bubbleSort(nums) {
    let hasSwaped;

    for (num of nums) {
        hasSwaped = false;
        for (let j = 0; j < nums.length; j++) {
            if (nums[j] > nums[j + 1]) {
                hasSwaped = true;
                [nums[j], nums[j + 1]] = [nums[j + 1], nums[j]];
            }
        }
        if (!hasSwaped) return nums;
    }
    return nums;
}

let nums = [7, 3, 2, 5, 6, 10, 9, 8, 1];
console.log(bubbleSort(nums));

File: Sorting/Bubble Sort/BubbleSort.cpp
#include <iostream>

using namespace std;

void bubbleSort(int (&array)[], int n)
{
    bool hasSwaped;
    for (int i = 0; i < n; i++)
    {
        hasSwaped = false;
        for (int j = 0; j < n - 1; j++)
        {
            if (array[j] > array[j + 1])
            {
                swap(array[j], array[j + 1]);
                hasSwaped = true;
            }
        }
        if (!hasSwaped)
            break;
    }

    // Print values
    for (int i = 0; i < n; i++)
    {
        cout << array[i] << " ";
    }

    return;
}

int main()
{

    int numbers[] = {7, 3, 2, 5, 6, 10, 9, 8, 1};
    int n = sizeof(numbers) / sizeof(numbers[0]);

    bubbleSort(numbers, n);

    return 0;
}